src/osp/Active/SysPrefabInit.cpp

```diff
@@ -0,0 +1,201 @@
+/**
+ * Open Space Program
+ * Copyright © 2019-2022 Open Space Program Project
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#include "SysPrefabInit.h"
+#include "SysRender.h"
+#include "SysHierarchy.h"
+
+#include <osp/Resource/resources.h>
+#include <osp/Resource/ImporterData.h>
+
+#include <Magnum/Trade/Trade.h>
+#include <Magnum/Trade/PbrMetallicRoughnessMaterialData.h>
+
+using Corrade::Containers::ArrayView;
+
+using osp::restypes::gc_importer;
+
+using osp::phys::EShape;
+
+namespace osp::active
+{
+
+void SysPrefabInit::init_hierarchy(
+        PrefabInitView_t                    prefabInits,
+        Resources const&                    rResources,
+        acomp_storage_t<ACompHierarchy>&    rHier) noexcept
+{
+    for (TmpPrefabInit const& rPrefab : prefabInits)
+    {
+        auto &rImportData = rResources.data_get<osp::ImporterData const>(
+                gc_importer, rPrefab.m_importerRes);
+        auto &rPrefabData = rResources.data_get<osp::Prefabs const>(
+                gc_importer, rPrefab.m_importerRes);
+
+        auto const objects = rPrefabData.m_prefabs[rPrefab.m_prefabId];
```

Comment:
you're copying this container onto the stack, but you only ever use it to access the .size().

Do `auto const& objects` instead?

Or even `size_t const numObjects = rPrefabData.m_prefabs[rPrefab.m_prefabId].size();`